File: exercises/quarkus/shipping-service/src/main/java/com/example/shipping/domain/model/ShipmentId.java
package com.example.shipping.domain.model;

import java.util.Objects;
import java.util.UUID;

public record ShipmentId(String value) {

    private static final String PREFIX = "ship_";

    public ShipmentId {
        Objects.requireNonNull(value, "ShipmentId value");
        if (value.isBlank()) {
            throw new IllegalArgumentException("ShipmentId value must not be blank");
        }
    }

    public static ShipmentId generate() {
        return new ShipmentId(PREFIX + UUID.randomUUID());
    }

    @Override
    public String toString() {
        return value;
    }
}

File: exercises/dotnet/ShippingService/Domain/Models/ShipmentId.cs
namespace ShippingService.Domain.Models;

/// <summary>
/// Value object wrapping a shipment identifier.
/// </summary>
public record ShipmentId(string Value)
{
    private const string Prefix = "ship_";

    public static ShipmentId Generate() => new($"{Prefix}{Guid.NewGuid()}");

    public override string ToString() => Value;
}
